x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/import/calculate_rule_source_for_import.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  PrebuiltRuleToImport,
+  RuleSource,
+} from '../../../../../../common/api/detection_engine';
+import type { PrebuiltRuleAsset } from '../../../prebuilt_rules';
+import { calculateRuleSourceFromAsset } from '../detection_rules_client/mergers/rule_source/calculate_rule_source_from_asset';
+
+/**
+ * Calculates the rule_source field for a rule being imported
+ *
+ * @param rule The rule to be imported
+ * @param prebuiltRuleAssets A list of prebuilt rule assets, which may include
+ * the installed version of the specified prebuilt rule.
+ * @param installedRuleIds A list of prebuilt rule IDs that are currently installed
+ */
+export const calculateRuleSourceForImport = ({
+  rule,
+  prebuiltRuleAssets,
+  installedRuleIds,
+}: {
+  rule: PrebuiltRuleToImport;
+  prebuiltRuleAssets: PrebuiltRuleAsset[];
+  installedRuleIds: string[];
+}): RuleSource => {
+  const matchingAsset = prebuiltRuleAssets.find((asset) => asset.rule_id === rule.rule_id);
```

Comment:
So, took me a while to wrap my head around this one. Correct me if I'm wrong:

```
  prebuiltRuleAssets, // contains only assets that match rule_id and version
  installedRuleIds, // contains all assets that match rule_id (version might be different to imported rule.version)
```

If that's right, small nit: rename `matchingAsset` with `assetWithMatchingVersion`

Reply:
Your understanding is correct (except for referring to `installedRuleIds` as "assets").

I made the changes you proposed (in b47153225492d5fcaa5b703d14fb3e7248a03ee6) as I do think it's an improvement, but there's still some implicit terminology here where we define "asset" to mean "matching on rule_id," which I don't think is totally intuitive.